pkg/ccl/changefeedccl/kvfeed/kv_feed.go

```diff
@@ -592,11 +618,15 @@ func (f *kvFeed) runUntilTableEvent(ctx context.Context, resumeFrontier span.Fro
 	}()
 
 	// We have catchup scan checkpoint.  Advance frontier.
-	if startFrom.Less(f.checkpointTimestamp) {
-		for _, s := range f.checkpoint {
-			if _, err := resumeFrontier.Forward(s, f.checkpointTimestamp); err != nil {
-				return err
-			}
+	if f.spanLevelCheckpoint != nil || startFrom.Less(f.checkpointTimestamp) {
```

Comment:
perhaps instead of having `if new checkpoint not nil` checks everywhere, you could coerce the old checkpoint into the new format if the new one isnt set, and use only new-style logic afterwards everywhere in the code.

like `if spanlevelcheckpoint == nil, spanlevelcheckpoint = map{oldcheckpointts: [oldchkpoint_spans...]`

thoughts?

Reply:
Reached out to you on slack regarding this point. Not sure if I understand the benefits of this proposal.